tests/HomeWork.spec.js

```diff
@@ -1,110 +1,130 @@
-    });
-
-    // Send file by mail
-    await documentPage.getUploadDocumentCheckBox.click();
-    await documentPage.getMoreLinks.click();
-    await documentPage.getSendLink.hover();
-    await documentPage.getSendEmailLink.click();
-    await page.waitForTimeout(9000)
+    }); 
+  });
+  
+  await test.step('Send file by mail', async () => 
+  {
+    await documentPage.clickUploadDocumentCheckBox();
+    await documentPage.clickMoreLinks();
+    await documentPage.getSendLinkover();
+    await documentPage.clickSendEmailLink();
+  });
 
+  await test.step('Fill email field and send mail', async () => 
+  {
     await mailPage.fillEmailField(data.email);
     await mailPage.getSubjectField.type(data.mailtext);
-    await page.waitForTimeout(3000)
-    expect(await mailPage.getAttachment.isVisible()).toBe(true);
+    await expect(mailPage.getAttachment, 'Attachment should be Visible').toBeVisible();
     await mailPage.getMailSendButton.click();
+  });
 
-    // Check mail is received
-    await page.waitForTimeout(9000)
+
+  await test.step('Check mail is received', async () => 
+  {
+    await page.waitForTimeout(6000);
     await mailPage.getRefreshMail.click();
     await mailPage.getRefreshMail.click();
     await mailPage.getMailTitle.click();
-    await page.waitForTimeout(3000)
-    expect(await mailPage.getAttachment.isVisible()).toBe(true);
-    await page.waitForTimeout(5000)
+    await page.waitForTimeout(3000);
+    await expect(mailPage.getAttachment, 'Attachment should be Visible').toBeVisible();
     await mailPage.getAttachment.hover();
     await mailPage.getAttachmentArrow.click();
     await mailPage.getSaveLink.click();
     await mailPage.getDocumentFolder.click();
     await page.waitForTimeout(2000);
     await mailPage.getSaveButton.click();
+  });
 
-    // Check mail is saved in documents
-    await page.waitForTimeout(2000);
-    await documentPage.getDocumentTab.click();
+  await test.step('Check mail is saved in documents', async () => 
+  {
+    await documentPage.clickDocumentTab();
     await page.waitForTimeout(2000);
-    expect(await documentPage.getSavedDocument.isVisible()).toBe(true);
+    await documentPage.visibleSavedDocument();
     await page.waitForTimeout(5000);
+  });
 
+  await test.step('Drag and drop document', async () => 
+  {
     await documentPage.dragAndDropDocument();
-    await documentPage.getCheckBoxAll.click();
+    await documentPage.clickCheckBoxAll();
     await page.waitForTimeout(5000);
     await documentPage.dragAndDrop();
     await page.waitForTimeout(2000);
-    expect(await documentPage.getEmptyDocumentText.isVisible()).toBe(true);
+    await documentPage.visibleEmptyDocumentText();
+  });
 
-    // Check document is in trash
+  await test.step('Check document is in trash', async () => 
+  {
     await page.waitForTimeout(2000);
-    await documentPage.getTrash.click();
+    await documentPage.clickTrash();
+    await page.waitForTimeout(3000);
```

Comment:
It's really better to not use `waitForTimeout` since it can make tests flaky. You can check next guide reagarding that: https://playwright.dev/docs/actionability#assertions. Also, `locator.waitFor` can be enough in general cases when you wait for some UI. But if you need something exceptional - try to use `expect.poll`: https://playwright.dev/docs/test-assertions#expectpoll.